### PR TITLE
Migrate category images

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Autoupgrade PR for https://github.com/PrestaShop/PrestaShop/pull/36877
| Type?             | 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Sponsor company   | 
| How to test?      |

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Hi Daniel, I think you may aim for the `dev` branch as this where we have all the changes for PrestaShop 9. If in the future we need them on the 6.x branch, we will manage all the back-porting.

**Comment by Hlavtox** [comment]
@Quetzacoalt91 Hi, understood, rebased, branch changed. :-)

File: upgrade/php/ps_900_migrate_category_images.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+     *
+     * So you suddenly got one thumbnail that was from a different image.
+     * Even worse, frontoffice and backoffice generated it as a different thumbnail,
+     * one as small, one as medium.
+     */
+    $formattedNameSmall = ImageType::getFormattedName('small');
+    $formattedNameMedium = ImageType::getFormattedName('medium');
+
+    // Get categories on shop
+    $categoryIds = Db::getInstance()->executeS('SELECT id_category FROM `' . _DB_PREFIX_ . 'category`');
```
Comment (M0rgan01) [inline upgrade/php/ps_900_migrate_category_images.php]:
```suggestion
    $categoryIds = DbWrapper::executeS('SELECT id_category FROM `' . _DB_PREFIX_ . 'category`');
```

It has a new DbWrapper class which allows error handling during the update.

Comment (Hlavtox) [reply upgrade/php/ps_900_migrate_category_images.php]: @M0rgan01 Suggestion applied. :-)

**Review by M0rgan01 — DISMISSED** [review]

**Comment by kpodemski** [comment]
@Hlavtox you didn’t import the DbWrapper class correctly

**Comment by Hlavtox** [comment]
@kpodemski All green now

**Comment by AureRita** [comment]
Hi @Hlavtox 

Can you resolve your conflict ? it seems to be here because of the validation of your previous PR. 

Waiting for feedback

**Comment by Hlavtox** [comment]
@AureRita Ready to go. :-)

Ping @ShaiMagal for reapproval

**Review by ShaiMagal — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @Hlavtox 

Thank you for your PR, I tested it and I see that different things works : 

if we only have a cover, we'll have a thumbnail 

[recording(29).webm](https://github.com/user-attachments/assets/d4a1cb06-19dc-42f0-bf40-22e8316c6688)

if we only have a thumbnail, it 'll delete the small part of them 

[recording(28).webm](https://github.com/user-attachments/assets/e917fcb4-68fd-498b-ae03-b56f527eecf9)

and if we have both of them, the small part will be deleted

[recording(27).webm](https://github.com/user-attachments/assets/ea08fbf1-19b1-4f4a-87a2-4de979a89f09)

but we can also see that it doesn't delete " Menu thumbnails ". Did you want to delete them in this PR on in another one ? 

Except this part, for me it can be a QA :heavy_check_mark: 

Tested on : 
8.0.4 to 8.2.0
8.0.4 to 9.0.0
8.2.0 to 9.0.0


Waiting for feedback

**Pull request merged**